src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs

```diff
@@ -0,0 +1,198 @@
+            {
+                { entityName, entity }
+            };
+
+            RuntimeEntities runtimeEntities = new(entities);
+            OpenApiDocument openApiDocument = await OpenApiTestBootstrap.GenerateOpenApiDocument(
+                runtimeEntities: runtimeEntities,
+                configFileName: CUSTOM_CONFIG,
+                databaseEnvironment: MSSQL_ENVIRONMENT);
+
+            foreach (OpenApiPathItem pathItem in openApiDocument.Paths.Values)
+            {
+                foreach ((OperationType operationType, OpenApiOperation operation) in pathItem.Operations)
+                {
+                    if (operationType is OperationType.Get)
+                    {
+                        Assert.IsTrue(operation.Parameters.Count is 5);
+
+                        // Assert that it contains all the query parameters with appropriate type
+                        Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name is "$select" && param.Schema.Type is "string"));
+                        Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name is "$filter" && param.Schema.Type is "string"));
+                        Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name is "$orderby" && param.Schema.Type is "string"));
+                        Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name is "$first" && param.Schema.Type is "integer"));
+                        Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name is "$after" && param.Schema.Type is "string"));
+                    }
+                    else
+                    {
+                        // No query parameters for other OperationTypes. 
+                        Assert.IsFalse(operation.Parameters.Any(param => param.In is ParameterLocation.Query));
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Validates that Query parameters are correctly generated for Stored Procedures.
+        /// Query Parameters for stored procedure will be generated for irrespective of the Operation Type.
+        /// If a parameter default value is not provided in the config, it will marked as required.
+        /// </summary>
+        /// <param name="entityName"></param>
+        /// <param name="objectName"></param>
+        /// <returns></returns>
+        [DataTestMethod]
+        [DataRow("CountBooks", "count_books", DisplayName = "StoredProcedure with no parameters")]
+        [DataRow("UpdateBookTitle", "update_book_title", DisplayName = "StoredProcedure with parameters")]
+        public async Task ValidateQueryParametersForStoredProcedures(string entityName, string objectName)
+        {
+            Dictionary<string, object> parameterDefaults = null;
```

Comment:
Is null not by default?

Reply:
i don't think so, because i was getting error to assign it before using.